File: GolangSolutions/BackTracking/M784_字母大小写全排列.go
package BackTracking

import "unicode"

func letterCasePermutation(s string) (res []string) {
	n := len(s)
	var backtrack func([]byte, int)
	backtrack = func(s []byte, idx int) {
		for idx < n && unicode.IsDigit(rune(s[idx])) {
			idx++
		}
		if idx == n {
			res = append(res, string(s))
			return
		}
		s[idx] ^= 32
		backtrack(s, idx+1)
		s[idx] ^= 32
		backtrack(s, idx+1)
	}
	backtrack([]byte(s), 0)

	return
}

File: CPPSolutions/LeetCode/BackTracking/M784_字母大小写全排列.cpp
#include <bits\stdc++.h>

using namespace std;

class Solution {
public:
    vector<string> letterCasePermutation(string s) {
        vector<string> res;
        backtrack(s, 0, res);

        return res;
    }

    void backtrack(string &s, int idx, vector<string> &res) {
        while (idx < s.size() && isdigit(s[idx])) {
            idx++;
        }
        if (idx == s.size()) {
            res.push_back(s);
            return;
        }
        s[idx] ^= 32;
        backtrack(s, idx + 1, res);
        s[idx] ^= 32;
        backtrack(s, idx + 1, res);
    }
};